jax/_src/scipy/linalg.py

```diff
@@ -1041,3 +1041,49 @@ def toeplitz(c: ArrayLike, r: Optional[ArrayLike] = None) -> Array:
       (nrows,), (1,), 'VALID', dimension_numbers=('NTC', 'IOT', 'NTC'),
       precision=lax.Precision.HIGHEST)[0]
   return jnp.flip(patches, axis=0)
+
+
+_NULLSPACE_DOC = """\
+Because the size of the output of ``null_space`` is data-dependent, the function is not
+typically compatible with JIT. The JAX version adds the optional ``size`` argument which
+must be specified statically for ``jax.scipy.linalg.null_space`` to be used within some of JAX's
+transformations.
+"""
+_NULLSPACE_EXTRA_PARAMS = """
+size : int, optional
+    If specified, ``size`` elements of the null space are returned. If there are
+    fewer null space elements than ``size`` indicates, the return value will be padded with constant
+    ``fill_value`` vectors.
+fill_value : array_like, optional
+    When ``size`` is specified and there are fewer null space elements than the indicated number of elements,
+    the return value will be padded with constant ``fill_value`` vectors. If ``fill_value`` is not specified,
+    it defaults to zero (indeed constant 0 vectors are always in the null space).
+"""
+
+@_wraps(scipy.linalg.null_space, lax_description=_NULLSPACE_DOC, extra_params=_NULLSPACE_EXTRA_PARAMS)
+def null_space(A: ArrayLike, rcond: Optional[float] = None,
+               size: Optional[int] = None, fill_value: Union[None, ArrayLike, Tuple[ArrayLike]] = None) -> Array:
+  u, s, vh = svd(A, full_matrices=True)
+  M, N = u.shape[0], vh.shape[1]
+  if rcond is None:
+    rcond = jnp.finfo(s.dtype).eps * max(M, N)
+  tol = jnp.amax(s) * rcond
+  num = jnp.sum(s > tol, dtype=int)
+  if size is None:
+    size = int(num)
+  else:
+    size = min(size, vh.shape[0])
+  if fill_value is None:
+    fill_value = 0
+  elif isinstance(fill_value, tuple):
+    fill_value = jnp.asarray(fill_value)
+  else:
+    fill_value = jnp.asarray(fill_value)
+  Q = lax.dynamic_slice_in_dim(vh, size, vh.shape[0] - size, axis=0)
+  # now we need to make sure to remove all vectors in Q that are not in the null space
+  # but were selected just because we need an array of size `size`
+  filler = jnp.ones_like(Q) * fill_value
+  mask = (jnp.arange(Q.shape[0]) < size-num)[:, None]
+  Q = jnp.where(mask, filler, Q)
```

Comment:
I think using `fill_value` directly in place of `filler` would be give the same result

Reply:
I think you are right, will correct

Reply:
done